### ISSUE TITLE
ArrayIndexOutOfBoundsException occurs seldomly.

### ISSUE THREAD

**Issue by TraceLosu** [Nathat23/StackMob-5#207]
```
[StackMob] Task #219590 for StackMob v5.5.3 generated an exception
java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 9
        at java.util.Arrays$ArrayList.set(Arrays.java:4356) ~[?:?]
        at net.minecraft.server.v1_16_R3.NonNullList.set(SourceFile:53) ~[patched_1.16.5.jar:git-Tuinity-"94f025a"]
        at net.minecraft.server.v1_16_R3.TileEntityLootable.setItem(TileEntityLootable.java:137) ~[patched_1.16.5.jar:git-Tuinity-"94f025a"]
        at org.bukkit.craftbukkit.v1_16_R3.inventory.CraftInventory.setItem(CraftInventory.java:100) ~[patched_1.16.5.jar:git-Tuinity-"94f025a"]
        at uk.antiperson.stackmob.listeners.ShearListener.lambda$onShearSheep$0(ShearListener.java:62) ~[?:?]
        at org.bukkit.craftbukkit.v1_16_R3.scheduler.CraftTask.run(CraftTask.java:100) ~[patched_1.16.5.jar:git-Tuinity-"94f025a"]
        at org.bukkit.craftbukkit.v1_16_R3.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:468) ~[patched_1.16.5.jar:git-Tuinity-"94f025a"]
        at net.minecraft.server.v1_16_R3.MinecraftServer.b(MinecraftServer.java:1475) ~[patched_1.16.5.jar:git-Tuinity-"94f025a"]
        at net.minecraft.server.v1_16_R3.DedicatedServer.b(DedicatedServer.java:419) ~[patched_1.16.5.jar:git-Tuinity-"94f025a"]
        at net.minecraft.server.v1_16_R3.MinecraftServer.a(MinecraftServer.java:1388) ~[patched_1.16.5.jar:git-Tuinity-"94f025a"]
        at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1120) ~[patched_1.16.5.jar:git-Tuinity-"94f025a"]
        at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:290) ~[patched_1.16.5.jar:git-Tuinity-"94f025a"]
```

This exception occurs in my server every so often, can't seem to find what it entails/what the cause of it.

Running StackMob 5.5.3 with Tuinity version git-Tuinity-"94f025a" (MC: 1.16.5) (Implementing API version 1.16.5-R0.1-SNAPSHOT).

**Comment by Nathat23** [comment]
From that error its to do with using dispensers to shear, and it seems that for some reason the item used to shear cannot be found in the dispenser?

**Comment by Nathat23** [comment]
Fixed in afbb00451afbcaa44045f648d0f984d5e3d983fe. You can download the latest build from https://ci.codemc.io/job/Nathat23/job/StackMob-5/ to fix.
